lib/util/git/index.ts

```diff
@@ -283,8 +283,8 @@ export async function syncGit(): Promise<void> {
       await git.raw(['remote', 'set-url', 'origin', config.url]);
       await resetToBranch(await getDefaultBranch(git));
       const fetchStart = Date.now();
-      await git.pull();
-      await git.fetch();
+      await git.fetch({ '--force': 'true' });
+      await git.pull({ '--rebase': 'true' });
```

Comment:
If the current default branch cannot "fast forward", shouldn't we be happy that it fails and a clone is done instead? Is there any scenario where a pull with rebase is desirable?

Reply:
Would be good.
However, there are no any errors or anything reported in log file while the renovate fetching repo, issues come when it tries to push or merge.
If you like I can attach full log (very long log ;-) ).

Reply:
if you worry about flags, I can start with swapping fetch with pull (fetch before pull) and see if this would help

Reply:
Fetch before pull seems safer. I don't want rebase until there's a reproduction to demonstrate, plus we're satisfied that there's no problems it causes

Reply:
the PR was updated, please have a look